File: StudioZaneeMusic(Mobile)/lib/models/product.dart
class Product {
  final int id;
  final String name;
  final double price;
  final String? description;
  final String imageUrl;
  final String categoryName;
  final bool isUnderMaintenance;
  final bool isRented;

  Product({
    required this.id,
    required this.name,
    required this.price,
    this.description,
    required this.imageUrl,
    required this.categoryName,
    required this.isUnderMaintenance,
    required this.isRented,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json["id"],
      name: json["name"] ?? "",
      price: (json["price"] ?? 0).toDouble(),
      description: json["description"],
      imageUrl: json["imageUrl"] ?? "",
      categoryName: json["categoryName"] ?? "Không rõ",
      isUnderMaintenance: json["isUnderMaintenance"] == 1,
      isRented: json["isRented"] == 1,
    );
  }
}

File: StudioZaneeMusic(Web)/Models/Product.cs
using System.ComponentModel.DataAnnotations;

namespace DuAnBai3.Models
{
    public class Product
    {
        public int Id { get; set; }

        [Required(ErrorMessage = "Tên sản phẩm là bắt buộc")]
        [StringLength(100)]
        public string Name { get; set; } = string.Empty;

        [Range(1000, 1000000000, ErrorMessage = "Giá phải lớn hơn 1000đ")]
        public decimal Price { get; set; }

        // ✅ Cho phép NULL trong DB
        public string? Description { get; set; }

        // ✅ Cho phép NULL trong DB
        public string? ImageUrl { get; set; }

        // ✅ Bỏ Required, vì DB cho phép null
        [Display(Name = "Danh mục")]
        public int? CategoryId { get; set; }

        public Category? Category { get; set; }

        // ✅ Có giá trị mặc định (false)
        public bool IsUnderMaintenance { get; set; } = false;

        public bool IsRented { get; set; } = false;
    }
}
